tests/ha/hawk_gui.pm

```diff
@@ -121,6 +123,9 @@ sub run {
 
     # Synchronize with the nodes
     barrier_wait("HAWK_GUI_CHECKED_$cluster_name");
+
+    # Wait for master node to reboot
+    barrier_wait("HAWK_FENCE_$cluster_name");
```

Comment:
Same comment here for the dual `barrier_wait`.

I think I see your point but calling `barrier_wait` two times one after the others without doing anything else (from the code PoV) is weird. Maybe something can be changed here?

Reply:
I need the first barrier to let the nodes know that the hawk_test app is finished, but the second barrier to wait for both nodes to reboot.